app/uk/gov/hmrc/eoricommoncomponent/frontend/controllers/auth/AuthAction.scala

```diff
@@ -47,6 +52,14 @@ class AuthAction @Inject() (
   private val baseRetrievals     = ggEmail and credentialRole and affinityGroup
   private val extendedRetrievals = baseRetrievals and internalId and allEnrolments and groupIdentifier and credentials
 
+  def enrolledUserClearingCacheOnCompletionAction(requestProcessor: RequestProcessorSimple) =
+    cacheClearOnCompletionAction.async(implicit request => authorise(requestProcessor))
+
+  def enrolledUserWithSessionAction(service: Service)(requestProcessor: RequestProcessorSimple) = {
+    val filter = new CompletedJourneyFilter(service, sessionCache, parser)
```

Comment:
I was thinking this, but I think my issue is that because services case objects extending a shared sealed trait and instead the Service object can technically be a lot more varied, being really rigid instead of accepting one in could lead to issues down the line if whatever pattern matching we do (or if we use a map here) doesn't match perfectly - such as with new services etc.

Let's have a chat if you have something in mind though!